data/index.js

```diff
@@ -139,7 +155,23 @@ export function select( reducerKey ) {
 		return select( reducerKey )[ selectorKey ]( ...args );
 	}
 
-	return selectors[ reducerKey ];
+	const createResolver = ( selector, key ) => ( ...args ) => {
+		const result = selector( ...args );
+		const resolver = resolvers[ reducerKey ] && resolvers[ reducerKey ][ key ];
+		if ( ! resolver ) {
+			return result;
+		}
+		const isFulfilled = resolver.isFulfilled( stores[ reducerKey ].getState(), ...args );
+		if ( isFulfilled ) {
+			// TODO: clear memize cache for these sepecific args
+		} else {
+			resolver.fulfill( ...args );
```

Comment:
How do we signal to the original consumer that the value is now ready after this completes?

Reply:
Since the selector is state based, the resolver will update the state at somepoint which will trigger a subscribe and rerender the consumer.